tracer/src/Datadog.AutoInstrumentation.NativeLoader/instrumented_assembly_generator/instrumented_assembly_generator_cor_profiler_function_control.cpp

```diff
@@ -0,0 +1,89 @@
+#include "instrumented_assembly_generator_cor_profiler_function_control.h"
+#include "instrumented_assembly_generator_helper.h"
+#include "instrumented_assembly_generator_metadata_interfaces.h"
+#include "../log.h"
+
+namespace instrumented_assembly_generator
+{
+InstrumentedAssemblyGeneratorCorProfilerFunctionControl::InstrumentedAssemblyGeneratorCorProfilerFunctionControl(
+    ICorProfilerFunctionControl* corProfilerFunctionControl, ICorProfilerInfo12* corProfilerInfo12, ModuleID moduleId,
+    mdMethodDef methodId) :
+    m_pICorProfilerFunctionControl(corProfilerFunctionControl),
+    m_corProfilerInfo(corProfilerInfo12),
+    m_moduleId(moduleId),
+    m_methodId(methodId)
+{
+    Log::Debug("InstrumentedAssemblyGeneratorCorProfilerFunctionControl::.ctor");
+    AddRef();
+}
+
+InstrumentedAssemblyGeneratorCorProfilerFunctionControl::~InstrumentedAssemblyGeneratorCorProfilerFunctionControl()
+{
```

Comment:
no call to release ?

Reply:
After ebde2f57c8ef03fd167862f981abd5c4958c222a, we hold the object as a `std::shared_ptr` in `CorProfiler` for the lifetime of the `CorProfiler`.
The [members of the class](https://github.com/DataDog/dd-trace-dotnet/pull/2829/files#diff-40c61227f2885c8d62bc372c1a49e9a204ff3b37dc5f4eb876df51680761546fR14) are defined as a `ComPtr` or `std::shared_ptr` and so their destructors (which calls Release()) will be called automatically so as I understand no need to call release here.
I tried to add a release call but it cause the process to crash, which seems to be good indication that this would be one decrement too much. WDYT?

Reply:
ok good one.